pupil_src/shared_modules/marker_tracker_3d/optimization/model_state.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+
+import numpy as np
+
+import file_methods
+from marker_tracker_3d import utils
+
+logger = logging.getLogger(__name__)
+
+
+class ModelState:
+    def __init__(self, save_path):
+        self.save_path = save_path
+
+        self.current_frame_id = 0
+
+        self.frames_id = []
+        self.markers_id = []
+
+        self.all_novel_markers = []
+
+        self.camera_extrinsics_opt = {}  # {frame id: camera extrinsics(list)}
```

Comment:
From these names its imo not clear that they are dicts (I saw the names a few times in your code and always thought they were lists). For dicts, I really like this naming scheme `keys_to_values`, examples:
```
frame_to_camera_extrinsics
frame_to_camera_extr
frame_id_to_camera_extr
marker_to_extrinsics
marker_id_to_extrinsics
```
This scheme makes clear 1) that it is a dict (if you know the scheme) and 2) what the keys and values are.

If you don't like the scheme for your case, you can also use e.g. `camera_extrinsics_dict`